components/script/dom/document.rs

```diff
@@ -2029,14 +2029,18 @@ impl Document {
         *self.asap_scripts_set.borrow_mut() = vec![];
         self.asap_in_order_scripts_list.clear();
         self.deferred_scripts.clear();
+        let global_scope = self.window.upcast::<GlobalScope>();
+        if self.loader.borrow_mut().cancel_all_loads() || global_scope.close_event_sources() {
+            // If any loads were canceled.
+            self.salvageable.set(false);
+        };
 
-        // TODO: https://github.com/servo/servo/issues/15236
-        self.window.cancel_all_tasks();
+        self.window.cancel_all_tasks_from_source(TaskSourceName::Networking);
```

Comment:
This carries an assumption that all the tasks queued on the networking task source are fetches. I think it is ok for now, but we'd definitely want to leave a comment here clarifying the assumption. Also, step annotations on this algorithm would be nice.

Reply:
Good catch, I'll add a note.

Reply:
I've also filed https://github.com/whatwg/html/issues/3837